Mailspring uses dpkg-architecture to detect the multiarch library path, but that doesn't work on non-Debian-based distributions, like our freedesktop SDK.
Using gcc -print-multiarch instead is portable and gives the same output.

Mailsync wants to statically link against a lot of libs, because it fears different versions between buildtime and runtime.
This is not a problem for flatpak, because we always have the exact same version.
Freedesktop SDK also doesn't ship static libraries, so this saves us from having to build them ourselves.

diff --git a/mailsync/CMakeLists.txt b/mailsync/CMakeLists.txt
index 62f1e82..69de024 100644
--- a/mailsync/CMakeLists.txt
+++ b/mailsync/CMakeLists.txt
@@ -7,9 +7,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 
 SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
 
-#Set Linker flags
-set(CMAKE_EXE_LINKER_FLAGS "-static-libgcc -static-libstdc++")
-
 
 find_package (PkgConfig REQUIRED)
 pkg_check_modules (GLIB REQUIRED glib-2.0)
@@ -22,7 +19,7 @@ IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
   # Detect Debian/Ubuntu multiarch library path (e.g. /usr/lib/aarch64-linux-gnu)
   # so that find_library() works on ARM64 and other non-x86 architectures
   execute_process(
-    COMMAND dpkg-architecture -qDEB_HOST_MULTIARCH
+    COMMAND gcc -print-multiarch
     OUTPUT_VARIABLE DEB_HOST_MULTIARCH
     OUTPUT_STRIP_TRAILING_WHITESPACE
     ERROR_QUIET
